### DISCUSSION TITLE
Best Practices: How to enable internationalization for your WordPress Code

### DISCUSSION THREAD

**Discussion by bph** [WordPress/developer-blog-content#30] [Topic Ideas]
Thank you to @zzap for suggesting this topic. 
This could cover both, backend and frontend.

**Comment by colorful-tones** [comment]
Some areas that this might include (in my mind):

- Using a text domain in a theme or plugin, e.g. `_e`, `_n`
- Loading translations [`load_theme_textdomain()`](https://developer.wordpress.org/reference/functions/load_theme_textdomain/), [load_child_theme_textdomain()](https://developer.wordpress.org/reference/functions/load_child_theme_textdomain/)
- Using `@wordpress/i18n` in JavaScript and `_n`, `_x`, `__`
- [wp_set_script_translations()](https://developer.wordpress.org/reference/functions/wp_set_script_translations/)
- [Create translation file](https://developer.wordpress.org/block-editor/how-to-guides/internationalization/#create-translation-file)

I'm sure I'm missing some things.

**Comment by bph** [comment]
I just saw: 
[Internationalization Functions](https://developer.wordpress.org/apis/internationalization/internationalization-functions/)

**Comment by zzap** [comment]
This is also under docs team effort to move all the info to one central place. https://github.com/WordPress/Documentation-Issue-Tracker/issues/624

**Comment by bcworkz-wp** [comment]
I see two possible approaches. Either table this idea until the other docs are better organized; or present this soon clarifying the current state of documentation.

In the support forums, I've noticed confusion between the do your own translation approach and the let polyglots do the translations. Could use some clarification? (not sure why anyone would want to do their own, but they do)
